src/compositions/useSyncedRecordString.ts

```diff
@@ -0,0 +1,47 @@
+import { Ref, ref, watch } from 'vue'
+import { isValidJsonString, stringify } from '@/utilities/json'
+
+export type UseSyncedRecordValue = Record<string, unknown>
+
+export type UseSyncedRecordString = {
+  valid: Ref<boolean>,
+  stringRef: Ref<string>,
+  recordRef: Ref<UseSyncedRecordValue>,
+}
+
+/**
+ * The useSyncedRecordString composition takes a record or record-parsable string
+ * and creates two reactive references: a string and an object.
+ * It then sets up watchers to sync changes between these two references.
+ * @param initialValues UseSyncedRecordValue
+ * @returns UseSyncedRecordString
+ */
+export function useSyncedRecordString(initialValues?: UseSyncedRecordValue | string): UseSyncedRecordString {
```

Comment:
Thoughts on renaming this `useJsonRecord`? And having it accept/return 

- valid
- json
- record